lib/internal/Magento/Framework/Code/Reader/ClassReader.php

```diff
@@ -17,46 +21,62 @@ class ClassReader implements ClassReaderInterface
      *
      * @param  string $className
      * @return array|null
-     * @throws \ReflectionException
+     * @throws ReflectionException
      */
     public function getConstructor($className)
     {
-        $class = new \ReflectionClass($className);
+        $class = new ReflectionClass($className);
         $result = null;
         $constructor = $class->getConstructor();
         if ($constructor) {
             $result = [];
-            /** @var $parameter \ReflectionParameter */
+            /** @var $parameter ReflectionParameter */
             foreach ($constructor->getParameters() as $parameter) {
                 try {
                     $result[] = [
                         $parameter->getName(),
-                        $parameter->getClass() !== null ? $parameter->getClass()->getName() : null,
+                        $this->getParameterClass($parameter),
                         !$parameter->isOptional() && !$parameter->isDefaultValueAvailable(),
                         $this->getReflectionParameterDefaultValue($parameter),
                         $parameter->isVariadic(),
                     ];
-                } catch (\ReflectionException $e) {
+                } catch (ReflectionException $e) {
                     $message = sprintf(
                         'Impossible to process constructor argument %s of %s class',
                         $parameter->__toString(),
                         $className
                     );
-                    throw new \ReflectionException($message, 0, $e);
+                    throw new ReflectionException($message, 0, $e);
                 }
             }
         }
 
         return $result;
     }
 
+    /**
+     * Get class by reflection parameter
+     * 
+     * @param ReflectionParameter $reflectionParameter
+     * @return ReflectionClass|null
+     * @throws ReflectionException
+     */
+    private function getParameterClass(ReflectionParameter $reflectionParameter): ?ReflectionClass
```

Comment:
It would be good to add the same fix to other occurrences of parameter class retrieval.

i.e. 
\Magento\Framework\Code\Generator\EntityAbstract::extractParameterType
\Magento\Framework\Interception\Code\InterfaceValidator::getParametersType

Reply:
Processed all files where found this function. 

Thanks!